### PR TITLE
Fix subfolders for backwards compatibility

### PR DESCRIPTION
**Pull request by x4m**
Recent refactoring changed subfolder hierarchy, this commit brings back old folder trees.

### CODE REVIEW

File: internal/s3_folder.go [hunk 1]
```diff
@@ -65,10 +66,7 @@ func (folder *S3Folder) ReadObject(objectRelativePath string) (io.ReadCloser, er
 }
 
 func (folder *S3Folder) GetSubFolder(subFolderRelativePath string) StorageFolder {
-	if !strings.HasSuffix(subFolderRelativePath, "/") {
-		subFolderRelativePath = subFolderRelativePath + "/"
-	}
-	return NewS3Folder(folder.uploader, folder.S3API, *folder.Bucket, folder.Path+subFolderRelativePath)
+	return NewS3Folder(folder.uploader, folder.S3API, *folder.Bucket, path.Join(folder.Path, subFolderRelativePath) + "/")
```
Comment (Tinsane) [inline internal/s3_folder.go]: Is usage of path.Join valid here? What happens here if we use some other platform, like Windows?

File: internal/backup.go [hunk 2]
```diff
@@ -34,7 +34,7 @@ type Backup struct {
 }
 
 func NewBackup(folder StorageFolder, name string) *Backup {
-	return &Backup{folder, name}
+	return &Backup{folder.GetSubFolder(BaseBackupPath), name}
```
Comment (Tinsane) [inline internal/backup.go]: It doesn't work for some backup creation scenarios. E. g. in HandleDelete. Check for this change correctness in all usages of NewBackup function.

File: internal/s3_folder.go (hunk 1 above)
Comment (x4m) [reply internal/s3_folder.go]: I've replaced path.join with hand-written, possibly more buggy, but platfrom-independent join

File: internal/backup.go (hunk 2 above)
Comment (x4m) [reply internal/backup.go]: Done.

**Pull request merged**